docs/05_operations/02_commands/04_genx-commands.md

```diff
@@ -0,0 +1,398 @@
+---
+title: 'Operations - Genx commands'
+sidebar_label: 'Genx commands'
+id: genx-commands
+keywords: [operations, genx, commands]
+tags:
+    - genx
+    - cli
+    - commands
+    - terminal
+---
+<!-- Genx  -->
+
+Genesis provides a `genx` which is a CLI scaffolding tool that enables you to create Genesis applications from scratch. This interface allows you to create a new project based on some existing seeds, so you do not need to worry about creating a new project from scratch.
+
+The `genx` has several options available to create, configure or modify the project. these are the available parameters:
+
+|Command | Argument      | Description|
+|--------|---------------|-------------|
+| [init](#init)   |`<app_Name>`          |   Create a new project|
+| [analyze](#analyze) |[folder]       |Analyze production bundle|
+| [clean](#clean)   |[...paths]     | Delete specified paths (defaults to dist folder)|
+| [build](#build)  |[folder]       | Build production bundle|
+| [dev](#dev)    |[folder]       |   Start development server|
+| [run](#run)    |`<task>` [module]|  Run monorepo task|
+| [serve](#serve)  |[folder]       |Start static file server (defaults to dist folder)|
+| [test](#test)   |[folder]       | Run tests|
+| [lint](#lint)   |[folder]       | Lint TypesSript and style files|
+| [upgrade](#upgrade) | [folder]      |Upgrade Foundation UI dependencies|
+
+## How to use it
+
```

Comment:
There are two main ways to run the Genx CLI:

* Standalone - outside of an existing project such as when scaffolding a new project
* Local - within an existing project such as when executing builds within a project

### Standalone

```
npx -y @genesislcap/genx@latest <command> [args]
```

For example, to scaffold a new project:

```
npx -y @genesislcap/genx@latest init myapp
```

This will use the latest available version of the CLI and create a new application called *myapp* using default seed *blank-app-seed*.

### Local

Once you have a project, you can add Genx CLI as a local dependency to your *package.json*:

```json
  "devDependencies": {
    ...
    "@genesislcap/genx": "...",
    ...
  },

```

You can find out the latest available version by running `npm info @genesislcap/genx` at any point.

Once you have bootstrapped NPM dependencies, you will be able to call Genx CLI in *package.json* scripts section like so:

```
"scripts": {
    "build": "genx <command> [args]"
}
```

For example:

```json
  "scripts": {
    "build": "genx build -e ENABLE_SSO",
    "dev": "genx dev -e API_HOST,ENABLE_SSO",
    "serve": "genx serve",
    "test": "genx test",
  },
```

Now, if you execute `npm run build` or `npm run dev` it will launch Genx CLI installed in your project locally.